Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+TypeSignatureDisambiguation.swift

```diff
@@ -0,0 +1,437 @@
+    }
+    
+    @inlinable
+    func union(_ other: Self) -> Self {
+        .init(storage: storage | other.storage)
+    }
+    
+    @inlinable
+    func intersection(_ other: Self) -> Self {
+        .init(storage: storage & other.storage)
+    }
+    
+    @inlinable
+    func symmetricDifference(_ other: Self) -> Self {
+        .init(storage: storage ^ other.storage)
+    }
+    
+    @inlinable
+    mutating func formUnion(_ other: Self) {
+        storage |= other.storage
+    }
+    
+    @inlinable
+    mutating func formIntersection(_ other: Self) {
+        storage &= other.storage
+    }
+    
+    @inlinable
+    mutating func formSymmetricDifference(_ other: Self) {
+        storage ^= other.storage
+    }
+}
+
+extension _TinySmallValueIntSet: Sequence {
+    func makeIterator() -> Iterator {
+        Iterator(set: self)
+    }
+    
+    struct Iterator: IteratorProtocol {
+        typealias Element = Int
+        
+        private var storage: UInt64
+        private var current: Int = -1
+        
+        @inlinable
+        init(set: _TinySmallValueIntSet) {
+            self.storage = set.storage
+        }
+        
+        @inlinable
+        mutating func next() -> Int? {
+            guard storage != 0 else {
+                return nil
+            }
+            // If the set is somewhat sparse, we can find the next element faster by shifting to the next value.
+            // This saves needing to do `contains()` checks for all the numbers since the previous element.
+            let amountToShift = storage.trailingZeroBitCount + 1
+            storage >>= amountToShift
+            
+            current += amountToShift
+            return current
+        }
+    }
+}
+
+// MARK: Table
+
+/// A fixed-size grid of elements.
+private struct Table<Element> {
+    typealias Size = (width: Int, height: Int)
+    @usableFromInline
+    let size: Size
+    private let storage: ContiguousArray<Element>
+
+    @inlinable
+    init(width: Int, height: Int, initializingWith initializer: (_ buffer: inout UnsafeMutableTableBufferPointer) throws -> Void) rethrows {
+        size = (width, height)
+        let capacity = width * height
+        storage = try .init(unsafeUninitializedCapacity: capacity) { buffer, initializedCount in
+            var wrappedBuffer = UnsafeMutableTableBufferPointer(width: width, wrapping: buffer)
+            try initializer(&wrappedBuffer)
+            initializedCount = capacity
+        }
+    }
+
+    struct UnsafeMutableTableBufferPointer {
+        private let width: Int
+        private var wrapping: UnsafeMutableBufferPointer<Element>
+
+        init(width: Int, wrapping: UnsafeMutableBufferPointer<Element>) {
+            self.width = width
+            self.wrapping = wrapping
+        }
+
+        @inlinable
+        func initializeElementAt(row: Int, column: Int, to element: Element) {
+            wrapping.initializeElement(at: index(row: row, column: column), to: element)
```

Comment:
Clever trick to wrap the underlying storage and make it appear as it it were a 2-D array.